pkg/kubelet/cm/dra/manager.go

```diff
@@ -83,9 +73,9 @@ func (m *ManagerImpl) PrepareResources(pod *v1.Pod) error {
 				claimName := resourceclaim.Name(pod, &pod.Spec.ResourceClaims[i])
 				klog.V(3).InfoS("Processing resource", "claim", claimName, "pod", pod.Name)
 
-				if claimInfo := m.cache.get(claimName, pod.Namespace); claimInfo != nil {
```

Comment:
It feels strange to me to hang all of these functions off of the state type. I would expect the state type to do exactly one thing -- act as the state store for any checkpointed information. The cache should then be a type in this package that may or may not extend the state type directly, that can then add this additional functionality if desired.